BasicGates/Workbook_BasicGates.ipynb

```diff
@@ -1521,16 +1633,176 @@
   },
   {
    "cell_type": "markdown",
+   "id": "679a9a71",
    "metadata": {},
    "source": [
     "[Return to Task 2.3 of the Basic Gates kata](./BasicGates.ipynb#Task-2.3.-Two-qubit-gate---3)."
    ]
   },
   {
    "cell_type": "markdown",
+   "id": "601a94e2",
    "metadata": {},
    "source": [
-    "## Task 2.4. Toffoli gate\n",
+    "## Task 2.4. Two-qubit gate - 4\n",
+    "\n",
+    "**Input:** Two unentangled qubits (stored in an array of length 2) in an arbitrary two-qubit state ${\\color{blue}\\alpha} |00\\rangle + {\\color{blue}\\beta} |01\\rangle + \\gamma |10\\rangle + \\delta |11\\rangle$.\n",
+    "\n",
+    "\n",
+    "**Goal:**  Change the two-qubit state to ${\\color{red}\\beta} |00\\rangle + {\\color{red}\\alpha} |01\\rangle + \\gamma |10\\rangle + \\delta |11\\rangle$.\n",
+    "\n",
+    "> This task can be solved using one library function; as an exercise, try to express the solution using several (possibly controlled) Pauli gates.  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5d48b697",
+   "metadata": {},
+   "source": [
+    "### Solution 1 - using intrinsic gates\n",
+    "\n",
+    "In vector form the transformation we need is:\n",
+    "\n",
+    "$$\n",
+    "\\begin{bmatrix}\n",
+    "   \\color{blue}\\alpha\\\\\n",
+    "   \\color{blue}\\beta\\\\\n",
+    "   \\gamma\\\\\n",
+    "   \\delta\\\\\n",
+    "\\end{bmatrix}\n",
+    "\\rightarrow\n",
+    "\\begin{bmatrix}\n",
+    "   \\color{red}\\beta\\\\\n",
+    "   \\color{red}\\alpha\\\\\n",
+    "   \\gamma\\\\\n",
+    "   \\delta\\\\\n",
+    "\\end{bmatrix}\n",
+    "$$\n",
+    "\n",
+    "This can be represented by a matrix:\n",
+    "$$\n",
+    "T = \\begin{bmatrix}\n",
```

Comment:
Maybe we can use $U$ to represent this transformation. Because we have a `T gate` available in Q#. So, it might make it a little difficult for somebody who is aware of that gate to follow along this task.

Although its not widely used like $X$, $H$ gates. Here is [link](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.t) to the documentation of T gate.

```suggestion
    "U = \\begin{bmatrix}\n",
```

Reply:
Thanks for catching this! I was planning to introduce the gate as the **T**arget gate, but ended up not including this explanation.